includes/admin/class-wc-rest-payments-files-controller.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Class WC_REST_Payments_Files_Controller
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for files.
+ */
+class WC_REST_Payments_Files_Controller extends WC_Payments_REST_Controller {
+
+	/**
+	 * Endpoint path.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'payments/file';
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'upload_file' ],
+				'permission_callback' => [ $this, 'check_permission' ],
+			]
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<file_id>\w+)',
+			[
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => [ $this, 'get_file' ],
+				'permission_callback' => [],
+			]
+		);
+	}
+
+	/**
+	 * Create file and respond with file object via API.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 */
+	public function upload_file( $request ) {
+		return $this->forward_request( 'upload_evidence', [ $request ] );
+	}
+
+	/**
+	 * Retrieve a file content via API.
+	 *
+	 * @param WP_REST_Request $request - request object.
+	 *
+	 * @return WP_Error|WP_HTTP_Response
+	 */
+	public function get_file( WP_REST_Request $request ) {
+		$file_id = $request->get_param( 'file_id' );
+		$result  = $this->forward_request( 'get_file_contents', [ $file_id ] );
+
+		if ( $result instanceof WP_Error ) {
+			$error_status_code = 'resource_missing' === $result->get_error_code() ? WP_Http::NOT_FOUND : WP_Http::INTERNAL_SERVER_ERROR;
+			return rest_ensure_response(
+				new WP_Error(
+					$result->get_error_code(),
+					$result->get_error_message(),
+					[ 'status' => $error_status_code ]
+				)
+			);
+		}
+
+		/**
+		 * WP_REST_Server will convert the response data to JSON prior to output it.
+		 * Using this filter to prevent it, and output the data from WP_HTTP_Response instead.
+		 */
+		add_filter(
+			'rest_pre_serve_request',
+			function ( bool $served, WP_HTTP_Response $response ) : bool {
+				echo $response->get_data(); // @codingStandardsIgnoreLine
+				return true;
+			},
+			10,
+			2
+		);
+
+		return new WP_HTTP_Response(
```

Comment:
Will this send out the response with the Content-Disposition header? If not, wouldn't that be better?

Reply:
Following @anu-rock's comment, I wonder if it would make sense to add a parameter to be able to specify the value of the Content-Disposition header? I mean to set it as `attachment` or `inline`

Reply:
by default, Content-Disposition is considered inline, but for clarity I just added also the content disposition header.